test/lib/analyzer/php.spec.ts

```diff
@@ -0,0 +1,15 @@
+import { phpFilesToScannedProjects } from "../../../lib/analyzer/applications";
+import * as drupal10FilePathToContents from "../../fixtures/php/drupal10FilePathToContent.json";
+
+const NUMBER_OF_INVALID_FILES_IN_DRUPAL_10 = 2;
+describe("Can create dependency tree when some files are invalid", () => {
+  it("Should succeed and return scan results", async () => {
+    const scanResults = await phpFilesToScannedProjects(
+      drupal10FilePathToContents,
+    );
+    expect(scanResults[0].facts[0].data._depPkgsList.length).toEqual(
+      Object.keys(drupal10FilePathToContents).length +
+        NUMBER_OF_INVALID_FILES_IN_DRUPAL_10,
+    );
```

Comment:
what exactly are we testing here? I don't have context here, but just from reading the test I have no idea why this is the expected outcome

Reply:
That's fair. At first, I just had a check to see if the function throws an error, but I thought that wasn't good enough.

I decided I wanted to ensure that only the specific problematic files were skipped

Reply:
So I'm comparing the number of files sent to the function with the number of dependencies received. They should be the same, except for the 2 problematic files.

Reply:
Well, the result already has 2 removed. I could change it to `result + 2 = original` as opposed to how I do it now which is `result = original - 2`

Reply:
that makes sense, but I think you wrote the test as `result = original + 2` (`+` instead of `-`)?

Reply:
You are absolutely right. The test was not correct from the get go. I fixed it and now it's much simpler.